src/PerfView/PerfViewData.cs

```diff
@@ -5729,10 +5732,27 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                         allocs = GetHeap(data.HeapHandle, heaps, ref lastHeapAllocs, ref lastHeapHandle);
                                     }
 
+                                    if (allocationSites == null)
+                                    {
+                                        var symReader = GetSymbolReader(log, SymbolReaderOptions.CacheOnly);
```

Comment:
The addresses/type names need to be extracted from each pdb before HeapAlloc events are processed. This implies that if a native pdb is resolved later, we need to process the trace again. I'm not sure how this should be handled. Maybe this should be in LoadSymbolsForModule?

Reply:
I think that the lookup of the AllocationSite info can be lazy. (as you encounter a new TraceModuleFile you add that information to 'allocationSites'.